pallets/parachain-staking/src/lib.rs

```diff
@@ -1930,26 +1595,85 @@ pub mod pallet {
 
 			Ok(Some(<T as pallet::Config>::WeightInfo::unlock_unstaked(unstaking_len)).into())
 		}
+
+		/// Claim block authoring rewards for the target address.
+		///
+		/// Requires `Rewards` to be set beforehand, which can by triggered by
+		/// any of the following options
+		/// * Calling increment_{collator, delegator}_rewards (active)
+		/// * Altering your stake (active)
+		/// * Leaving the network as a collator (active)
+		/// * Revoking a delegation as a delegator (active)
+		/// * Being a delegator whose collator left the network, altered their stake or incremented
+		///   rewards (passive)
+		///
+		/// The dispatch origin can be any signed one, e.g., anyone can claim
+		/// for anyone.
+		///
+		/// Emits `Rewarded`.
+		#[pallet::weight(<T as Config>::WeightInfo::claim_rewards())]
+		pub fn claim_rewards(origin: OriginFor<T>) -> DispatchResult {
+			let target = ensure_signed(origin)?;
+
+			// reset rewards
+			let rewards = Rewards::<T>::take(&target);
+			ensure!(!rewards.is_zero(), Error::<T>::RewardsNotFound);
+
+			// mint into target
+			let rewards = T::Currency::deposit_into_existing(&target, rewards)?;
+
+			Self::deposit_event(Event::Rewarded(target, rewards.peek()));
+
+			Ok(())
+		}
+
+		/// Actively increment the rewards of a collator.
+		///
+		/// The same effect is triggered by changing the stake or leaving the
+		/// network.
+		///
+		/// The dispatch origin must be a collator.
+		#[pallet::weight(<T as Config>::WeightInfo::increment_collator_rewards())]
+		pub fn increment_collator_rewards(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
Do I understand correct that if users want to claim their reward, they have to call this function first to update their reward?

Reply:
Yes, to do it manually. If someone leaves the group of delegators/nominators or modifies his stake, it will happen automatically...